Lib9c/Action/SecureMiningReward.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Immutable;
+using Bencodex.Types;
+using Lib9c.Abstractions;
+using Libplanet;
+using Libplanet.Action;
+using Libplanet.Assets;
+using Nekoyume.Model.State;
+
+namespace Nekoyume.Action
+{
+    [ActionType("secure_mining_reward")]
+    public class SecureMiningReward : ActionBase, ISecureMiningReward
+    {
+        // Copied from https://github.com/planetarium/lib9c/blob/v100361/Lib9c/Policy/BlockPolicySource.cs#L112-L118
+        // FIXME consider better way to check.
+        internal static readonly ImmutableHashSet<Address> AuthorizedMiners = new []
+        {
+            new Address("ab1dce17dCE1Db1424BB833Af6cC087cd4F5CB6d"),
+            new Address("3217f757064Cd91CAba40a8eF3851F4a9e5b4985"),
+            new Address("474CB59Dea21159CeFcC828b30a8D864e0b94a6B"),
+            new Address("636d187B4d434244A92B65B06B5e7da14b3810A9"),
+        }.ToImmutableHashSet();
+
+        public SecureMiningReward(Address miner, Address recipient, FungibleAssetValue amount)
+        {
+            Miner = miner;
+            Recipient = recipient;
+            Amount = amount;
+        }
+
+        public Address Miner { get; private set; }
+
+        public Address Recipient { get; private set; }
+
+        public FungibleAssetValue Amount { get; private set; }
+
+        public override IValue PlainValue => new List(
+            Miner.Bencoded,
+            Recipient.Bencoded,
+            Amount.Serialize()
+        );
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            IAccountStateDelta state = context.PreviousStates;
+            if (context.Rehearsal)
+            {
+                return state.MarkBalanceChanged(
+                    Amount.Currency,
+                    new [] { Miner, Recipient}
```

Comment:
Just style comment, you can ignore.

```suggestion
                    new [] { Miner, Recipient }
```